libmamba/include/mamba/util/graph.hpp

```diff
@@ -157,6 +164,14 @@ namespace mamba::util
         }
     };
 
+    template <typename Graph>
+    auto
+    is_reachable(const Graph& graph, typename Graph::node_id source, typename Graph::node_id target)
+        -> bool;
+
+    template <typename Graph, typename UnaryFunc>
+    void topological_sort_for_each_node_id(const Graph& graph, UnaryFunc&& func);
```

Comment:
Same remark regarding the name, do we really need the `for_each_node_id` part of the name?